java/system-test/system-test-suite/src/main/java/sleeper/systemtest/suite/dsl/query/SystemTestQueryResults.java

```diff
@@ -14,18 +14,31 @@
  * limitations under the License.
  */
 
-package sleeper.systemtest.suite.dsl;
-
-import com.amazonaws.services.s3.AmazonS3;
+package sleeper.systemtest.suite.dsl.query;
 
+import sleeper.core.record.Record;
 import sleeper.systemtest.drivers.instance.SleeperInstanceContext;
 import sleeper.systemtest.drivers.query.S3ResultsDriver;
+import sleeper.systemtest.drivers.query.WaitForQueryDriver;
+import sleeper.systemtest.suite.fixtures.SystemTestClients;
+
+import java.util.stream.Stream;
 
 public class SystemTestQueryResults {
     private final S3ResultsDriver s3ResultsDriver;
+    private final WaitForQueryDriver waitForQueryDriver;
+
+    public SystemTestQueryResults(SleeperInstanceContext instance, SystemTestClients clients) {
+        this.s3ResultsDriver = new S3ResultsDriver(instance, clients.getS3());
+        this.waitForQueryDriver = new WaitForQueryDriver(instance, clients.getDynamoDB());
+    }
+
+    public void waitForQuery(String queryId) throws InterruptedException {
+        waitForQueryDriver.waitForQuery(queryId);
+    }
 
-    public SystemTestQueryResults(SleeperInstanceContext instance, AmazonS3 s3) {
-        this.s3ResultsDriver = new S3ResultsDriver(instance, s3);
+    public Stream<Record> results(String queryId) {
+        return s3ResultsDriver.results(queryId);
     }
```

Comment:
These two new methods aren't needed. Since this just delegates to the driver, the driver can be used directly instead.

Since this class is part of the DSL, it should only expose methods which are steps in the DSL. Someone trying to use the DSL in a test would think they should be able to call these methods, but they can't because the query IDs are not exposed to tests.